File: nodes/message-bus/message-bus.process.py
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        preserve_order = settings.get('preserve_order', True)
        filter_empty = settings.get('filter_empty', True)

        # Helper function to extract messages from input
        def extract_messages(input_data):
            if not input_data:
                return []
            
            if isinstance(input_data, list):
                return input_data
            elif isinstance(input_data, dict) and 'role' in input_data:
                return [input_data]
            else:
                return []

        # Helper function to filter empty messages
        def filter_messages(messages):
            if not filter_empty:
                return messages
            
            filtered = []
            for message in messages:
                if not message:
                    continue
                if not isinstance(message, dict):
                    continue
                if 'role' not in message:
                    continue
                if message.get('content') is None:
                    continue
                if isinstance(message.get('content'), str) and message.get('content').strip() == '':
                    continue
                if isinstance(message.get('content'), list) and len(message.get('content')) == 0:
                    continue
                filtered.append(message)
            return filtered

        all_messages = []
        inputs_used = 0

        # Process inputs in order (input_1 through input_8)
        input_keys = ['input_1', 'input_2', 'input_3', 'input_4', 'input_5', 'input_6', 'input_7', 'input_8']
        
        for input_key in input_keys:
            input_data = inputs.get(input_key)
            messages = extract_messages(input_data)
            
            if len(messages) > 0:
                filtered_messages = filter_messages(messages)
                if len(filtered_messages) > 0:
                    all_messages.extend(filtered_messages)
                    inputs_used += 1

        return {
            "messages": all_messages,
            "message_count": len(all_messages),
            "inputs_used": inputs_used
        }

    except Exception as error:
        print(f'Message Bus error: {error}')
        raise Exception(f"Message Bus error: {str(error)}")

File: sdks/nodejs/nodes/message-bus/message-bus.process.js
export default async ({inputs, settings, config}) => {
    try {
        const preserveOrder = settings.preserve_order !== false;
        const filterEmpty = settings.filter_empty !== false;

        // Helper function to extract messages from input
        const extractMessages = (input) => {
            if (!input) return [];
            
            if (Array.isArray(input)) {
                return input;
            } else if (typeof input === 'object' && input.role) {
                return [input];
            } else if (typeof input === 'string') {
                // Convert string to user message
                return [{
                    role: 'user',
                    content: input
                }];
            } else {
                return [];
            }
        };

        // Helper function to filter empty messages
        const filterMessages = (messages) => {
            if (!filterEmpty) return messages;
            
            return messages.filter(message => {
                if (!message) return false;
                if (typeof message !== 'object') return false;
                if (!message.role) return false;
                if (message.content === undefined || message.content === null) return false;
                if (typeof message.content === 'string' && message.content.trim() === '') return false;
                if (Array.isArray(message.content) && message.content.length === 0) return false;
                return true;
            });
        };

        const allMessages = [];
        let inputsUsed = 0;

        // Process inputs in order (input_1 through input_8)
        const inputKeys = ['input_1', 'input_2', 'input_3', 'input_4', 'input_5', 'input_6', 'input_7', 'input_8'];
        
        for (const inputKey of inputKeys) {
            const input = inputs[inputKey];
            const messages = extractMessages(input);
            
            if (messages.length > 0) {
                const filteredMessages = filterMessages(messages);
                if (filteredMessages.length > 0) {
                    allMessages.push(...filteredMessages);
                    inputsUsed++;
                }
            }
        }

        return {
            messages: allMessages,
            message_count: allMessages.length,
            inputs_used: inputsUsed
        };

    } catch (error) {
        console.log('Message Bus error:', error);
        throw new Error(`Message Bus error: ${error.message}`);
    }
};
